Shorten Server guide (#725)
jamulussoftware/jamulussoftware.github.io: wiki/en/Running-a-Server.md

</details>

## Do I need to run a Server?

> **No**. You can use the Servers listed by the built in Directories and use Jamulus without running a Server or choose a third party hosting service. If you just want an undisturbed session, use the [soloing technique described on the Tips and Tricks page](tips-tricks-more#have-a-undisturbed-session-on-any-server).

Reviewer: > If you just want an undisturbed session, use the [soloing technique

I'm still not happy.  This should be a _last resort_ if no empty server can be found.  I don't think we should encourage people to take advantage of a resource provided to them for free by server operators who likely didn't intend small cliques to use the server to the exclusion of others - that being the nature of regstering a server with one of the built in directories.

> ## Basic requirements

Reviewer: Line 28 (old) now appears to have vanished.  I'd suggested (somewhere), moving under `Do I need to run a Server?`.

Author: Moved it to the Do I need to run a server section - although I'd be in favor of shortening it.

Reviewer: I think it's fine now.

> **No**. You can use the Servers listed by the built in Directories and use Jamulus without running a server or choose a third party hosting service to set up a Jamulus Server for you. If you just want an undisturbed session, use the [soloing technique described on the Tips and Tricks page](tips-tricks-more#have-a-undisturbed-session-on-any-server).

Reviewer: > and use Jamulus without running a server

That seems redundant somehow.  (And "server" should have been "Server".)

> or choose a third party hosting service

"or you can choose a third party hosting service".  Otherwise, the "You can" at the start should be removed.

> a third party hosting service to set up a Jamulus Server for you

Even shorter: "a third party Jamulus hosting service"

Reviewer: > If you just want an undisturbed session, ...

I'm not really comfortable having this here, still.  It still feels too much like sanctioning what I see as bad behaviour.

> ### Speed and latency

Reviewer: This is part of "Do I need to run a Server?" - that doesn't seem right.

Reviewer: Possibly add a new level 2 heading before this, maybe "Basic requirements".

**_The capability of the Server itself (and the network it's on) is NOT the main determinant of the quality of a Jamulus session!_**

Usually, problems are on the _Client_ side and should be fixed there. Have a look at the [Troubleshooting page](/wiki/Client-Troubleshooting) if needed.
Various problems can arise when setting up Servers - especially when run on a low-bandwidth home connection.
Therefore:
* Use other peoples' (public) Servers at first to isolate any issues that may in fact be due to Client configuration.
* Ensure you have enough bandwidth: Less than 5 players on a slower-speed home connection (eg 10 Mbit/s down and 1 Mbit/s up) are fine. You can read more about network requirements at [different quality settings here](Server-Bandwidth).

### General notes

> * Consider using a cloud host, not your home internet connection, to get better ping times if you're having problems

Reviewer: This _is_ answering "Do I need to run a Server?", so it belongs before the "Speed and Latency" section.

</details>

## Do I need to run a Server?

**No**. You can use the Servers listed by the built in Directories and use Jamulus without running a server.

> 

Reviewer: It probably warrants another level 2 section heading at this point (i.e. after line 23 (new)), given the new approach, having answered the question.

Author: See my new commit.

> **Note:** You can even have a "private", uninterrupted session with other people on a Server by simply soloing each other. You will then not be able to hear anyone else if they enter your Server (note that this does not stop them from hearing you, using the chat function, or seeing your profile information).

Reviewer: This shouldn't be as highlighted as it is.  It's an "emergency measure" only.  It's not good behaviour.

Reviewer: Possibly remove it entirely, as it's not about running a server - maybe under Client Hints and Tips, with a note that it's for times when there are no sutiable servers with no one on, really.

Author: @gilgongo Did you add this to "move people away" from setting up private servers? I think this should still be documented somewhere, but maybe massively shortened.

Reviewer 2: Move it to the FAQ?

Author: Or tips and tricks?

Reviewer 2: On second thoughts, yes - T&T. The FAQ says is really for more general stuff.

Reviewer: And link that from line 23.

Author: Ok. Please see my latest commits

If you don't want to use any of the Servers listed by the built in Directories, you may be able to use a Server (either Registered in a Custom Directory, or Unregistered - see [Server Types](#server-types)) hosted by a third party. Doing so will save you the trouble of setting one up yourself.

> 

Reviewer: I'd add line 30 (old) after line 27 (new).

Author: Hmm. Did that. But do we really need this line?

Reviewer: I'm assuming line 29 (new now) is what I was saying should be kept.  Yes, I do still think it's needed - it's another reason that the answer to "Do I need to run a server?" is "No".  Not everyone is aware of third party providers.

Author: Added the third party provider text back in a slightly condensed form. I hope that's ok?

### Speed and latency

**_The capability of the Server itself (and the network it's on) is NOT the main determinant of the quality of a Jamulus session!_**

Usually, problems are on the _Client_ side and should be fixed there.
Various problems can arise when setting up Servers - especially when run on a low-bandwidth home connection.
Therefore:

> * Use other peoples' (public) Servers at first to isolate any issues that may in fact be due to Client configuration.

Reviewer: Do we have a Client Troubleshooting page that can be linked here?

Author: Yes. Linked.

Reviewer: Linked on line 31 (new) and then we say the same thing again here.  I don't think it needs saying twice.

Author: Ok. See the upcoming update.

* Ensure you have enough bandwidth: Less than 5 players on a slower-speed home connection (eg 10 Mbit/s down and 1 Mbit/s up) are fine. You can read more about network requirements at [different quality settings here](Server-Bandwidth).

### In general

> * Consider using a cloud host, not your home internet connection to get better ping times if you're having problems

Reviewer: Comma after "connection".

Author: Fixed.

* Any Server should have at least 1.6GHz CPU frequency and 1GB RAM
* Running a Server may require you to adjust any firewalls running on or outside of your machine or cloud host.

> * Running an Uregistered Server at home will require you to [port forward](#running-an-unregistered-server) on your router. When running a Registered Server, port forwarding should not be necessary in most cases, but it's advisable to do so because some networks may not work properly with Jamulus in its default mode.

Reviewer: Elsewhere `port forward` has been linked directly to the section on Port Forwarding.  A consistent approach should be taken.

Author: Fixed.

> * Jamulus only has limited IPv6 support which needs to be enabled with a command line argument on the Client and Server. There are plans to expand IPv6 support.

Reviewer: Put "There are plans to expand IPv6 support." in brackets - it's only vaguely relevant to the discussion.

Author: Removed

## Server Types

You can run your Server in different ways (either at home or on a 3rd party host):

### 1. Registered

> Your Server will be listed in a Directory. By default, Jamulus has a list of Public Directories that Clients can connect to. If you register with one of these, anyone can then discover and connect to your Server. You can also have your Server listed on a Custom Directory, if that better meets your needs.

Reviewer: Mm, not pressing for this but "built in" rather than "Public" is my current preference...  "Built in" works against "Custom" better than "Public" against "Custom".

Author: Ok. Changed it. But I think it's not much different.

While setting up a Server isn't difficult, it's a good idea to read the following backround information to avoid some problems:

### Speed and latency

**_The capability of the Server itself (and the network it's on) is NOT the main determinant of the quality of a Jamulus session!_**

Usually, problems are on the _Client_ side and should be fixed there. Have a look at the [Troubleshooting page](/wiki/Client-Troubleshooting) if needed.
Various problems can arise when setting up Servers - especially when run on a low-bandwidth home connection.
Therefore:
* Use other peoples' (public) Servers at first to isolate any issues that may in fact be due to Client configuration.
* Ensure you have enough bandwidth: Less than 5 players on a slower-speed home connection (eg 10 Mbit/s down and 1 Mbit/s up) are fine. You can read more about network requirements at [different quality settings here](Server-Bandwidth).

### General notes

* Consider using a cloud host, not your home internet connection, to get better ping times if you're having problems

> * Any Server should have at least 1.6GHz CPU frequency and 1GB RAM

Reviewer: Given recent discussions, can we provide a chart _somewhere_ - ideally as part of this guide - saying roughly what bandwidth and CPU requirements for every N clients, based on "worst case" (stereo, high quality, fast update, small network buffers).  That way people have some idea what to expect from a Pi4 or an AWS instance, etc.

Reviewer 2: Bandwidth guidelines seem easy enough (we pretty much [have that here](https://jamulus.io/wiki/Running-a-Server#bandwidth-use) don't we?). But CPU use seems less clear. Have their been recent discussions about that?

Reviewer: Ah right, I'd forgotten about [the bandwidth table](https://jamulus.io/wiki/Running-a-Server#bandwidth-use).

The discussion arguing for PCM audio proposes that the compression/decompression CPU usage is a limiting factor in the number of clients a server can support to a greater extent than bandwidth.

Reviewer 2: OK, but do I recall some kind of benchmarky stuff on CPU that somebody once did? Could we collate that into something like cores vs clients vs load? Seems a bit of a rat's nest.

Reviewer: Yeah - now you mention it, I vaguely remember some graph stuff.  It was on one of the earlyish multithreading branches, I think.  I'm not sure now who was involved...  @softins do you remember?

Reviewer 3: > Yeah - now you mention it, I vaguely remember some graph stuff. It was on one of the earlyish multithreading branches, I think. I'm not sure now who was involved... @softins do you remember?

I'm afraid not, I would also have to go and search

Reviewer: Maybe we raise a separate issue to update the guide with the performance info.

Reviewer: OK, added https://github.com/jamulussoftware/jamuluswebsite/issues/729.

	<figcaption>How Unregistered Servers work</figcaption>
</figure>

If you are running an unregistered server behind a home internet connection, you might need to enable [port forwarding](#port-forwarding) as described below.

### 3. Custom Directory

If you want to run a number of Servers, possibly also behind a firewall or on a LAN, you may want to run your Server as a Directory. Examples include online events, music associations, sectional rehearsals or music lessons for schools.

> To run a Directory [read this guide](Custom-Directories)

Reviewer: > (Custom-Directories)

Maybe we can rename that to "Running-a-Directory"?

Reviewer 2: We could. Not a hugely good idea to have large numbers of redirecting files really though, and we already have quite a few.

Reviewer: Oh - is that doc already released?  I thought it was post-3.8.2.  Oh well.

> ## Installing and running a Server

Reviewer: The whole page is "Running a Server".  This should focus on "Installation" or "Installation and Configuration".

Author: Done.

> To run a Custom Directory [read this guide](Custom-Directories)

Reviewer: I'll say it again - I don't like the heading being "Custom Directory".  It's just "Directory".  It's what Volker does for Any Genre 1, 2 and 3 and what I do for the genre-specific Directories.  And Volker does for at least one other and I do for four others.  Mine run identically except for port number (and welcome message).

Author: Changed

> To run a Custom Directory [read this guide](Custom-Directories)

Author: ```suggestion
To run a Directory [read this guide](Custom-Directories)
```

> # Installing and running a Server

Reviewer: This should be a level 2 heading.

Author: ```suggestion
## Installing and running a Server
```

Author: Done in next commit.

Most people run Jamulus as a "pure" Server on **hardware without audio** (e.g. on a 3rd party/cloud host) running Linux. The following steps assume you are familiar with the command line and Debian/Ubuntu or similar distribution using systemd. To run a server on Windows or on the desktop with a graphical user interface, [see this section](#servers-on-the-desktop).  

If you want to run a Server on a Raspberry Pi, you will need to [compile from source](https://github.com/jamulussoftware/jamulus/blob/master/COMPILING.md). See also this [guide for Raspberry Pi](/kb/2020/03/28/Server-Rpi.html) maintained by Jamulus user fredsiva.

> ### Installation

Reviewer: The preceding heading level was 1, this is heading level 3.  Is that jump correct?

> 

Reviewer: Line 73 (new) should just be "Directory".

Author: Is this still valid? I thought I changed that.

Reviewer: "Central Server" changed to "Directory Server" and then "Directory Server" changed to "Directory".

Reviewer: Changed on line 73 (new) but the heading needs changing, too.

Author: Ok. Removed all Directory Server occurrences.

Reviewer: `Custom Directory` should still be just `Directory`.

1. Reload the systemd files `sudo systemctl daemon-reload` and restart the headless Server: `sudo systemctl restart jamulus-headless`
1. Check all is well with `systemctl status jamulus-headless` (hit `q` to get back to the command prompt).

> You can control Jamulus with the `systemctl` command. For example, to stop the Server cleanly:

Reviewer: Lines 92 to 94 (new) Still aren't installation.  I'd move this down into a "Running" section that covers both command line and desktop.  It'll need more than this, of course.

`sudo systemctl stop jamulus-headless`

_To upgrade your Server, just repeat the steps above._

> ### Configuration

Reviewer: Still none of this (lines 98 to 130 new) is configuration.  Merge it with all the other running bits (including desktop).

#### Running in Registered mode

> The following minimum setup is required to [run a Registered Server](Running-a-Server#server-types):

Reviewer: > to [run a Registered Server](Running-a-Server#server-types):

Only the server type should be linked: "to run a [Registered Server](Running-a-Server#server-types):"

Reviewer: I'm still not happy with lines 103-134.  As a _guide_, this should go into more depth about what the options mean, rather than give what looks like a "copy-n-paste this and run your server now!" command.

"Configuration" is where we should be explaining what each option is that affects how a server is run and how that option can be set, either from the command line or, when available, from the UI.  I really don't like the split between the two as things stand.

Reviewer 2: That might make it worse from an accessibility standpoint if it's not clear what the link will give you: https://usability.yale.edu/web-accessibility/articles/links#link-text

Reviewer: The link explains _what_ a Registered Server _is_ -- not _how to run_ a Registered Server.  That's why it shouldn't include "run a" in the link.

1. Reload the systemd files `sudo systemctl daemon-reload` and restart the headless Server: `sudo systemctl restart jamulus-headless`
1. Check all is well with `systemctl status jamulus-headless` (hit `q` to get back to the command prompt).

_To upgrade your Server, just repeat the steps above._

> If you are running an unregistered server behind a home internet connection, you might need to enable [port forwarding](port-forwarding) as described below.

Reviewer: Is the target for the link right?   (No `#`.)

Reviewer 2: "Enable the headless Server process via systemd" - is that actually necessary any more with the .deb? I think it might be left over from when you had to roll your own unit file - does the installer not enable it for you?

Author: Fixed the initial error. And I think enabling is necessary.

### Configuration

> You can control Jamulus with the `systemctl` command. For example, to stop the Server cleanly:

Reviewer: Lines 105 to 107 aren't _Configuration_.

Reviewer: In fact, none of 103 to 139 is giving any guidance on _Configuration_.

Author: Could you please give a suggestion? CLI? Administration? Controlling the server?

Reviewer 2: This stuff is currently in a general "Notes" section because I couldn't think of a better name for it. But maybe "Controlling the Server" is good enough.

Author: Changed to your suggestion, but I'm still not sure if it's the best way to explain this section. What is it actually about? Somewhat it's related to CLI commands a server admin should know.

Reviewer 2: How about: [now done a commit with the suggestions]

That will mean the TOC reads as:

- Installation
- Configuration
- Maintenance

Author: Can you add your proposal via GitHub's suggestion feature?

Reviewer 2: Hm. All I can work out how to do is edit the file - will that do?

Author: Yes. That's ok too: https://github.com/ann0see/jamuluswebsite/edit/feature/minimizeServerManual/wiki/en/Running-a-Server.md

Although a suggestion would be nicer - but not needed.

Reviewer 2: OK I did a commit to that file.

Reviewer 2: How come that commit never turned up?

Author: I think it did? Can you check if the diff if it changed. I squashed multiple commits

Reviewer 2: If I run the minimizeServerManual branch locally with jekyll, I can't see those changes (eg the heading still says "Controlling the Server").

It may be I don't know how to use git to get the right branch from your repo though (I don't know what a "feature branch" is)

Author: Did you use the zip from this PR? I can have a look at the md file again

Reviewer 2: No, I just` git clone` and `git checkout feature/minimizeServerManual` How do I get the zip? Do I just unpack that then?

Author: Ok. Try the zip from this PR and check it – maybe I wrongfully reverted your commit

Reviewer 2: The [zip from your repo](https://github.com/ann0see/jamuluswebsite/archive/refs/heads/feature/minimizeServerManual.zip), I assume?  I tried that but it didn't have the changes. I can do them again on https://github.com/ann0see/jamuluswebsite/edit/feature/minimizeServerManual/wiki/en/Running-a-Server.md unless there's some way of using GH to suggest the changes, but I can't see how.

Reviewer 2: OK I re-did it and can see the changes.

> 

Reviewer: Change needed on line 129 (new) for to drop "Custom".

### Maintenance

#### Viewing The Logs

Jamulus will log to the system log file if you left the `StandardOutput=journal` setting in the unit file.

To view the log, use `journalctl` (to exit press Ctrl-C). For example, to read the system log file, filtered for the Jamulus service:

> 

Reviewer: I'd change the "Recording" heading (line 141 new) to "Controlling Recording".

> 

Reviewer: (new) Line 137 conflicts with (new) line 92.

Author: Removed the line 137. However, Upgrading is still maintenance - but can probably be removed...

You can see the result of these commands if you run `systemctl status jamulus`, or by viewing the logs.

> ## Servers on the desktop

Reviewer: As noted, I'm not particularly happy with the split.

Author: I think this should be part of another PR.

Reviewer: Covered by https://github.com/jamulussoftware/jamuluswebsite/issues/729

You can see the result of these commands if you run `systemctl status jamulus`, or by viewing the logs.

> ### Making a Server status page

Reviewer: Moved to Hints and Tips?

Author: No. I think this should no longer be documented - especially as we have JSON-RPC around the corner.

Reviewer: JSON-RPC doesn't solve the problem this does.  This spits out HTML that can be used without further thought by anyone.  JSON-RPC needs a programmer to come along and write a program to send the request and parse the output and generate the HTML.

Until someone supplies the equivalent functionality for JSON-RPC, I think this should stay.

Reviewer 2: Another one for T&T then?

Author: Ok. Moved.

> # Servers on the desktop

Reviewer: Another level 2 heading.

Author: Fixed.

> Jamulus can be configured to run in Server mode on the desktop. This gives you a graphical user interface to control most of the settings.

Reviewer: > Jamulus can be configured to run in Server mode on the desktop.

Should read

"Jamulus can be run in Server mode from the desktop."

Author: Changed.

* **Windows users** - Use the "Jamulus Server" icon in the Windows Start menu.
* **macOS users** - Double-click the "Jamulus Server" icon in Applications (assuming you put the files from the install there as per [these instructions](Installation-for-Macintosh)).

> * **Linux users** - Launch the "Jamulus Server" shortcut. Or you can open a terminal window (`CTRL+ALT+t` on Debian and related distros), type `jamulus -s` and hit return. _(To run a headless Server [read this guide](#installing-and-running-a-server))_

Reviewer: > _(To run a headless Server [read this guide](#installing-and-running-a-server))_

That's a bit senseless here, as it's about the GUI version.

Reviewer 2: @pljones Please stop commenting - this isn't the version that was intended :-)

Author: Removed.

> 

Reviewer: Aha!!  What follows appears to start talking about configuration...

Reviewer 2: yeah, this whole PR is broken. Ignore/

Reviewer: Line 221 (new) `## Options` is a bit weird - L2 heading in the middle of a "list" of L3 headings.

Reviewer: The following still need organising and merging with the command line options to give a proper guide about configuring the server.

The
> \## Server Setup

and
> \## Options

headings here are the **tab names** on the UI - but the information is applicable to both UI and CLI instances.

Reviewer 2: > organising and merging with the command line options

I'd say that's a different PR to get the relevant CLUI options in among the GUI tab/screenshot stuff. Also implies we should do that for the client Software Manual too (but I hope we can keep the shared options in an include as today or we'll have FOUR places to keep them in sync) . 

@ann0see is just trying the shorted the overall thing here.

Author: > @ann0see is just trying the shorted the overall thing here.

Yes. I think another PR should be opened (@pljones could you please open an issue)

Reviewer: Done https://github.com/jamulussoftware/jamuluswebsite/issues/729

**Note**: When your Server is recording, Clients will display a message that recording is on.

> ### Directory Address

Reviewer: Should still say `Custom Directory address`, as it's the name of the field on the UI and should match exactly.

<figure><img src="{% include img/en-screenshots/server-active.inc %}" style="float:left; margin-right:10px;" loading="lazy" alt="Image of the Jamulus Server icon"></figure> The Server is occupied

## Command line options

Most common functions in Jamulus can be set using the GUI, but these and others can also be set using options given in a terminal window. Exactly how you do this will depend on your operating system.

For example on Windows, to use a specific settings file, right-click on the Jamulus shortcut and choose "Properties" > Target. Add the necessary arguments to Jamulus.exe:

```shell
 "C:\Program Files\Jamulus\Jamulus.exe" --serverbindip 192.168.0.100
```

For macOS, start a Terminal window and run Jamulus with the desired options like this:

```shell
 /Applications/Jamulus.app/Contents/MacOS/Jamulus --serverbindip 192.168.0.100
```

**Note**: Command line options will set the Server's defaults at startup. You can override them with their corresponding GUI controls while the Server is running.

{% include_relative Include-Server-Commands.md %}

{% include_relative Include-Shared-Commands.md %}

> ## Running an Unregistered Server

Reviewer: Merge this with all the other "Running" stuff (including desktop).

Reviewer: See https://github.com/jamulussoftware/jamuluswebsite/issues/729

It is highly recommended to test your Server on a **Public Directory** first so as to narrow down any subsequent problems in unregistered mode.

> ### Setting up a Server behind a home router

Reviewer: This can apply regardless of whether you're running an Unregistered or Registered Server or a Directory.  It shouldn't be under "Running an Unregistered Server".  It needs to be under "Configuration".

**Note:** The default port for the current version of Jamulus is **UDP** (not TCP) port **22124**. You will usually forward the port **22124** from outside your network to the port **22124** of the machine running the Server.

**Note:** Your home router may change the IP address of the machine you're running your Server on. Depending on your router, you might need to give this machine a static IP address (often under DHCP settings of your router).

#### Getting the external IP

To allow others to connect to your Server from the internet, get your external (WAN) IP address e.g. by [using Google](https://www.google.com/search?q=what+is+my+ip) and give it to them.
You yourself should connect using the local network (LAN) address of the machine the Server is running on. If you are running a Client on the same machine as your Server, that would be `localhost` or `127.0.0.1`.

#### Dynamic DNS and why you will probably need it

> Most domestic internet connections change their external IP address after a short period, you might want to set up a "dynamic DNS" which gives you a static (sub-)domain you can access and share with others. For further information, please look if your router already provides this feature or use an external DynamicDNS provider.

Reviewer 2: Don't you always have to use an external provider for dynamic DNS (unless you roll your own if you have your own named)? This is sort of why I'd prefer just removing this section. But, otherwise:

```suggestion
Most domestic internet connections change their external IP address after a short period. To avoid problems with this, you might want to set up "dynamic DNS" to get a static (sub-)domain you can share with others. Your router may have a built-in client you can use, or you can set one up on a computer on your network.
```

Author: Some routers support it (FritzBox by AVM) out of the Box; the manufacturer provides the DynamicDNS service. I think this is the first place to check. 

> To avoid problems with this

I don’t think there are many problems if you don’t have Dynamic DNS. It’s just very inconvenient without.

Reviewer 2: By "this" I meant "change their external IP address after a short period" - so the problem of suddenly nobody being able to connect to your server. I'd regard that as a problem myself, but if it's not, then why make a big deal about ddns? :-) 

But again, I think all this just involves us in an area we don't need to get mixed up in: what routers do or do not do, etc.

Reviewer: Most built in "Dynamic DNS" is generally just support for a certain set of external providers.  The router knows when it's had a new IP from the ISP.  So you set up the Dynamic DNS provider in the router (i.e. enter your registration details) and the router makes sure the provider gets an appropriate "ping" to update your address.

The only other way it could possibly work is if your host name is registered with a registrar who allows you to provide your own DNS and you can configure your router to update the address of that DNS server from some other IP address.  I've not seen a router with that capability, though.  Sort of a "roll your own Dynamic DNS".

Author: Yeah. I think the confusion is an edge case for FritzBoxes (Routers which are very common in Germany). I'll slightly modify the suggestion by @gilgongo

#### Getting the external IP

To allow others to connect to your Server from the internet, get your external (WAN) IP address e.g. by [using Google](https://www.google.com/search?q=what+is+my+ip) and give it to them.
You yourself should connect using the local network (LAN) address of the machine the Server is running on. If you are running a Client on the same machine as your Server, that would be `localhost` or `127.0.0.1`.

#### Dynamic DNS and why you will probably need it

Most domestic internet connections change their external IP address after a short period, you might want to set up a "dynamic DNS" which gives you a static (sub-)domain you can access and share with others. For further information, please look if your router already provides this feature or use an external DynamicDNS provider.

## Backing up the Server

_Note that headless Servers do not use `.ini` files. All configuration is given as command line options._

{% include_relative Include-Backing-Up.md %}

> ## Command line options

Reviewer: I'd move this section up before line 256 (new) "Running an Unregistered Server".

Author: Moved.

> 

Reviewer: Line 256 (new) needs to be a level 2 heading.

Author: Fixed.

### Port forwarding

If you are running a Server on your home network, people from outside will not be able to see things inside it. If you want to operate a Jamulus Server, you need to set up your router's firewall to let Jamulus Clients connect to the internal Server.

**Note:** The default port for the current version of Jamulus is **UDP** (not TCP) port **22124**.

The exact setup of port forwarding differs for every router. For help see [portforward.com](https://portforward.com).

Once you have your router set up, you can get your external (WAN) IP address e.g. by [using Google](https://www.google.com/search?q=what+is+my+ip). Give this address to your friends so they can connect to your Server (_but see also the note on dynamic DNS below_). You yourself should connect using the local network (LAN) address of the machine the Server is running on. If you are running a Client on the same machine as your Server, that would be `localhost` or `127.0.0.1`.

## Dynamic DNS and why you will probably need it

> Most domestic Internet connections will change their IP address after a period (hours, days or weeks). To make it easier for people to connect to you, you might want set up a "dynamic DNS address". You can do this on [the machine](https://www.online-tech-tips.com/computer-tips/ddns-dynamic-dns-service/) that you are running the Jamulus Server on, or preferably your router [may support it](https://www.noip.com/support/knowledgebase/how-to-configure-ddns-in-router/).

Reviewer: This para needs splitting.  It needs to say:
- What is "Dynamic DNS" (as that's what the heading has first)
- Why you will probably need it (as that's what the heading says the section explains)

Anything else (such as how to set it up) is extra - and should be separate (same section, different para).

Reviewer: This also doesn't stress that it's the _external_ IP address being discussed, not the LAN address.  Making that clear would make line 274 (new) easier to understand.

Reviewer 2: I'd be fine with just deleting this section on the grounds that it's not necessary to run Jamulus and is too long/involved.

Author: I would at least like to have a hint to dynamic dns (most people don't know about it).

Reviewer 2: OK, just a passing mention would be enough since "dynamic dns" will be easy easy enough to google for.

Author: Ok. See my edit.

Note also that your home router may also change the IP address of the machine that you are running your Server on. In which case you may need to give that machine a static IP in the router's DHCP configuration.

> 

Reviewer: Lines 276 (new) and 282 (new) both want to be level 2 headings.

Author: Fixed. Maybe it should even be a deeper level.

Reviewer: They came from a page describing the server UI and are literally the tab names on the UI, rather than section headings in a document.  All the headings refer literally to widgets on the UI.

However, that's not really appropriate here - this page isn't for describing the UI.  Hence it needs rewriting completely.

https://github.com/jamulussoftware/jamuluswebsite/issues/729

Reviewer 2: > All the headings refer literally to widgets on the UI.

As is the case for the client manual.

> Hence it needs rewriting completely.

I think (yet another) complete re-write is a separate PR once this PR has determined how to make things shorter.

> # Troubleshooting

Reviewer: Level 2 heading

Author: Done.

> If you plan to be playing regularly with the same people, **you are strongly advised** to first make sure that each member of the group is set up to use Jamulus properly. Do this by finding a Public Server with a reasonable ping time for all of you (20 ms or less perhaps), all connect to that and work to fix any individual issues (verifying that they can [follow Rule Number One](Client-Troubleshooting#you-all-sound-ok-but-its-difficult-to-keep-together) in particular). Use the solo technique above to prevent being interrupted if needed.

Reviewer: Given this para is almost word for word what was said earlier and the following two say nothing useful, this section could just have "See Server Troubleshooting" and nothing else.

Reviewer 2: I do feel bad about this... !

Author: Yeah. It's just a copy from above. Maybe we should move this to the Server troubleshooting page?

@gilgongo ?

Reviewer 2: The advice to use other people's Servers is to head off any issues later, so it's not really troubleshooting. If we want to say it at all, should be part of the setup really.

Author: I've removed it. I think there are valuable practical tips on that page, but probably too many...
